### ISSUE TITLE
Support smartcase for Sneak

### ISSUE THREAD

**Issue by kevintraver** [NeoVintageous/NeoVintageous#750]
Support smartcase in vim-sneak plugin

`let g:sneak#use_ic_scs = 1`

**Comment by gerardroche** [comment]
A new setting has been added:

```
    // 0: Always case-sensitive
    // 1: Case sensitivity is determined by 'ignorecase' and 'smartcase'.
    "vintageous_sneak_use_ic_scs": 0,
```

**Comment by gerardroche** [comment]
I just released 1.23.0. Blog post here: https://blog.gerardroche.com/2020/12/12/neovintageous-1.23.0/

Let me know if this implementation works for you.
